tools/cli/commands/release.js

```diff
@@ -150,12 +156,39 @@ export async function scriptRouter( argv ) {
 				  - You will now likely want to start a new release cycle like so:
 				      jetpack release ${ argv.project } new-cycle \n`.replace( /^\t+/gm, '' );
 			break;
-		case 'append':
-		case 'new-cycle':
+		case 'amend':
+			await checkBranchValid( argv );
+			argv.script = `vendor/bin/changelogger`;
+			argv.scriptArgs = [ `write`, `--amend` ];
+			argv.workingDir = `projects/${ argv.project }`;
+			argv.next = `Finished! Next:  
+				  	- You will now likely want to update readme.txt again, then commit to the release branch:
+				    	jetpack release ${ argv.project } readme \n`.replace( /^\t+/gm, '' );
+			break;
+		case 'version':
 			console.log( `${ argv.script } is not implemented yet!` );
 			process.exit( 1 );
 	}
 }
+
+/**
+ * Checks and makes sure we're on a release branch before we run the `amend` script.
+ *
+ * @param {object} argv - the arguments passed.
+ */
+export async function checkBranchValid( argv ) {
+	const currentBranch = child_process.execSync( 'git branch --show-current' ).toString().trim();
+	const branchPrefix = await readComposerJson( argv.project ).extra[ 'release-branch-prefix' ];
+	if ( ! currentBranch.match( `${ branchPrefix }/branch-` ) ) {
```

Comment:
Failure cases:

* If you try to amend on a project that doesn't have a release-branch-prefix set, it should probably give a different error reflecting that it can't amend.
* Also, if you checkout a branch named "undefined/branch-1.2.3", such a project shouldn't accept that as the release branch.
* You're using a regex here where you probably don't intend to, and in particular it's unanchored so a branch like "foobar-jetpack/branch-" would be accepted for Jetpack. What you probably want here is ``currentBranch.startsWith( `${ branchPrefix }/branch-` )``.

Reply:
One day, I will know when to use Regex. But it is not this day.

Let me know if that's better.

> Also, if you checkout a branch named "undefined/branch-1.2.3", such a project shouldn't accept that as the release branch.

I could explicitly check for such a thing, but not sure if that's necessary with the current logic?